File: VB/Q139895_2/main_form.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars

Namespace DXSample
	Public Class MainForm
		Inherits RibbonForm
		Implements IItemRefreshSupport
		Public Sub New()
			MyBase.New()
		 InitializeComponent()
		End Sub
		Public Sub InitializeComponent()
			Me.ribbon = New DevExpress.XtraBars.Ribbon.RibbonControl()
			Me.bbiNew = New DevExpress.XtraBars.BarButtonItem()
			Me.bbiClose = New DevExpress.XtraBars.BarButtonItem()
			Me.rpMain = New DevExpress.XtraBars.Ribbon.RibbonPage()
			Me.rpgActions = New DevExpress.XtraBars.Ribbon.RibbonPageGroup()
			Me.SuspendLayout()
			' 
			' ribbon
			' 
			Me.ribbon.ApplicationButtonKeyTip = ""
			Me.ribbon.ApplicationIcon = Nothing
			Me.ribbon.Items.AddRange(New DevExpress.XtraBars.BarItem() { Me.bbiNew, Me.bbiClose})
			Me.ribbon.Location = New System.Drawing.Point(0, 0)
			Me.ribbon.MaxItemId = 2
			Me.ribbon.Name = "ribbon"
			Me.ribbon.Pages.AddRange(New DevExpress.XtraBars.Ribbon.RibbonPage() { Me.rpMain})
			Me.ribbon.SelectedPage = Me.rpMain
			Me.ribbon.Size = New System.Drawing.Size(713, 143)
			' 
			' bbiNew
			' 
			Me.bbiNew.Caption = "New"
			Me.bbiNew.Id = 0
			Me.bbiNew.Name = "bbiNew"
'			Me.bbiNew.ItemClick += New DevExpress.XtraBars.ItemClickEventHandler(Me.bbiNew_ItemClick);
			' 
			' bbiClose
			' 
			Me.bbiClose.Caption = "Close"
			Me.bbiClose.Id = 1
			Me.bbiClose.Name = "bbiClose"
'			Me.bbiClose.ItemClick += New DevExpress.XtraBars.ItemClickEventHandler(Me.bbiClose_ItemClick);
			' 
			' rpMain
			' 
			Me.rpMain.Groups.AddRange(New DevExpress.XtraBars.Ribbon.RibbonPageGroup() { Me.rpgActions})
			Me.rpMain.KeyTip = ""
			Me.rpMain.Name = "rpMain"
			Me.rpMain.Text = "Main"
			' 
			' rpgActions
			' 
			Me.rpgActions.ItemLinks.Add(Me.bbiNew)
			Me.rpgActions.ItemLinks.Add(Me.bbiClose)
			Me.rpgActions.KeyTip = ""
			Me.rpgActions.Name = "rpgActions"
			' 
			' MainForm
			' 
			Me.ClientSize = New System.Drawing.Size(713, 558)
			Me.Controls.Add(Me.ribbon)
			Me.IsMdiContainer = True
			Me.Name = "MainForm"
			Me.Ribbon = Me.ribbon
			Me.Text = "DX Sample"
			Me.ResumeLayout(False)

		End Sub

		Private WithEvents bbiNew As DevExpress.XtraBars.BarButtonItem
		Private rpMain As RibbonPage
		Private rpgActions As RibbonPageGroup
		Private WithEvents bbiClose As DevExpress.XtraBars.BarButtonItem



		#Region "IItemRefreshSupport Members"

		Public Function GetCategory(ByVal text As String) As RibbonPageCategory Implements IItemRefreshSupport.GetCategory
			Return ribbon.MergedCategories(text)
		End Function
		Public Function GetPage(ByVal text As String) As RibbonPage Implements IItemRefreshSupport.GetPage
			Return ribbon.MergedPages(text)
		End Function
		Public ReadOnly Property IsMerged() As Boolean Implements IItemRefreshSupport.IsMerged
			Get
				Return False
			End Get
		End Property

		Public Function GetGroup(ByVal text As String) As RibbonPageGroup Implements IItemRefreshSupport.GetGroup
			For Each page As RibbonPage In ribbon.MergedPages
				For Each group As RibbonPageGroup In page.Groups
					If group.Text = text Then
						Return group
					End If
				Next group
			Next page
			Return Nothing
		End Function

		#End Region

		Private Sub bbiNew_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) Handles bbiNew.ItemClick
            Dim _editor As Editor = New Editor()
            _editor.MdiParent = Me
            _editor.Show()
		End Sub

		Private Sub bbiClose_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) Handles bbiClose.ItemClick
			Close()
		End Sub
	End Class
End Namespace

File: CS/Q139895_2/main_form.cs
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars;

namespace DXSample {
    public class MainForm :RibbonForm, IItemRefreshSupport {
        public MainForm() : base() { InitializeComponent(); }
        public void InitializeComponent() {
            this.ribbon = new DevExpress.XtraBars.Ribbon.RibbonControl();
            this.bbiNew = new DevExpress.XtraBars.BarButtonItem();
            this.bbiClose = new DevExpress.XtraBars.BarButtonItem();
            this.rpMain = new DevExpress.XtraBars.Ribbon.RibbonPage();
            this.rpgActions = new DevExpress.XtraBars.Ribbon.RibbonPageGroup();
            this.SuspendLayout();
            // 
            // ribbon
            // 
            this.ribbon.ApplicationButtonKeyTip = "";
            this.ribbon.ApplicationIcon = null;
            this.ribbon.Items.AddRange(new DevExpress.XtraBars.BarItem[] {
            this.bbiNew,
            this.bbiClose});
            this.ribbon.Location = new System.Drawing.Point(0, 0);
            this.ribbon.MaxItemId = 2;
            this.ribbon.Name = "ribbon";
            this.ribbon.Pages.AddRange(new DevExpress.XtraBars.Ribbon.RibbonPage[] {
            this.rpMain});
            this.ribbon.SelectedPage = this.rpMain;
            this.ribbon.Size = new System.Drawing.Size(713, 143);
            // 
            // bbiNew
            // 
            this.bbiNew.Caption = "New";
            this.bbiNew.Id = 0;
            this.bbiNew.Name = "bbiNew";
            this.bbiNew.ItemClick += new DevExpress.XtraBars.ItemClickEventHandler(this.bbiNew_ItemClick);
            // 
            // bbiClose
            // 
            this.bbiClose.Caption = "Close";
            this.bbiClose.Id = 1;
            this.bbiClose.Name = "bbiClose";
            this.bbiClose.ItemClick += new DevExpress.XtraBars.ItemClickEventHandler(this.bbiClose_ItemClick);
            // 
            // rpMain
            // 
            this.rpMain.Groups.AddRange(new DevExpress.XtraBars.Ribbon.RibbonPageGroup[] {
            this.rpgActions});
            this.rpMain.KeyTip = "";
            this.rpMain.Name = "rpMain";
            this.rpMain.Text = "Main";
            // 
            // rpgActions
            // 
            this.rpgActions.ItemLinks.Add(this.bbiNew);
            this.rpgActions.ItemLinks.Add(this.bbiClose);
            this.rpgActions.KeyTip = "";
            this.rpgActions.Name = "rpgActions";
            // 
            // MainForm
            // 
            this.ClientSize = new System.Drawing.Size(713, 558);
            this.Controls.Add(this.ribbon);
            this.IsMdiContainer = true;
            this.Name = "MainForm";
            this.Ribbon = this.ribbon;
            this.Text = "DX Sample";
            this.ResumeLayout(false);

        }

        private DevExpress.XtraBars.BarButtonItem bbiNew;
        private RibbonPage rpMain;
        private RibbonPageGroup rpgActions;
        private DevExpress.XtraBars.BarButtonItem bbiClose;
        private RibbonControl ribbon;



        #region IItemRefreshSupport Members

        public RibbonPageCategory GetCategory(string text) { return ribbon.MergedCategories[text]; }
        public RibbonPage GetPage(string text) { return ribbon.MergedPages[text]; }
        public bool IsMerged { get { return false; } }

        public RibbonPageGroup GetGroup(string text) {
            foreach (RibbonPage page in ribbon.MergedPages) foreach (RibbonPageGroup group in page.Groups) if (group.Text == text) return group;
            return null;
        }

        #endregion

        private void bbiNew_ItemClick(object sender, ItemClickEventArgs e) {
            Module module = new Module();
            module.MdiParent = this;
            module.Show();
        }

        private void bbiClose_ItemClick(object sender, ItemClickEventArgs e) { Close(); }
    }
}